File: python/svix/models/ingest_endpoint_out.py
# this file is @generated
import typing as t
from datetime import datetime

from .common import BaseModel


class IngestEndpointOut(BaseModel):
    created_at: datetime

    description: str
    """An example endpoint name."""

    disabled: t.Optional[bool] = None

    id: str
    """The Endpoint's ID."""

    metadata: t.Dict[str, str]

    rate_limit: t.Optional[int] = None

    uid: t.Optional[str] = None
    """Optional unique identifier for the endpoint."""

    updated_at: datetime

    url: str

File: kotlin/lib/src/main/kotlin/models/IngestEndpointOut.kt
// This file is @generated
package com.svix.kotlin.models

import kotlin.time.Instant
import kotlinx.serialization.Serializable

@Serializable
data class IngestEndpointOut(
    /** The Endpoint's ID. */
    val id: String,
    val url: String,
    val description: String,
    /**
     * Maximum messages per second to send to this endpoint.
     *
     * Outgoing messages will be throttled to this rate.
     */
    val throttleRate: UShort? = null,
    /** Optional unique identifier for the endpoint. */
    val uid: String? = null,
    val disabled: Boolean? = null,
    val createdAt: Instant,
    val updatedAt: Instant,
    val metadata: Map<String, String>,
)
